modules/tinymce/src/core/test/ts/browser/content/EditorContentTest.ts

```diff
@@ -80,28 +91,20 @@ describe('browser.tinymce.core.content.EditorContentTest', () => {
     assertEventsContentType();
   });
 
-  it('TINY-6281: getContent text', () => {
-    const editor = hook.editor();
-    editor.setContent('<p>Text to be retrieved</p>');
-    const text = editor.getContent({ format: 'text' });
-    // TODO: TINY-8367 The table plugin code has been moved to core so there is now always an extra DIV in the dom for the table resize bars
-    // Safari differs in behaviour compared to the other browsers when getting text content for inline mode.
-    // When the resize bar div is included in the DOM, editor.getBody().innerText includes two extra \n at the end
-    const isSafari = PlatformDetection.detect().browser.isSafari();
-    const expected = 'Text to be retrieved' + (isSafari ? '\n\n' : '');
-    Assertions.assertHtml('Should be expected text', expected, text);
-    assertEventsFiredInOrder();
-    assertEventsContentType();
-  });
+  // TODO: TINY-8367 The table plugin code has been moved to core so there is now always an extra DIV in the dom for the table resize bars
+  // Safari differs in behaviour compared to the other browsers when getting text content for inline mode.
+  // When the resize bar div is included in the DOM, editor.getBody().innerText includes two extra \n at the end
+  it('TINY-6281: getContent text', () => testGetTextContent('<p>Text to be retrieved</p>', 'Text to be retrieved' + (isSafari() ? '\n\n' : '')));
```

Comment:
I looked into this more and the cause appears to be that Safari adds on the newlines at the end because it's an inline editor that uses a `<div>` element as a wrapper. If you put this same content in an iframe editor (i.e as a child of the body) then you don't end up with the newlines. I guess on the bright side, that should mean these oddities shouldn't happen as often when using a "classic" editor.

I wonder if instead of trying to use `innerText` here we should make it share the same logic as the `selection.getContent()` API and use a range over all the content to get the text instead. So this is another option we could use to improve this further, so yeah I'd suggest logging a Jira and detail both this option and the option Andy mentioned as potential solutions.

- [ ] Log a follow up Jira to improve/fix these Safari oddities

Reply:
For tracking purposes, this was logged as TINY-8690.